go/pkg/client/tree_test.go

```diff
@@ -1573,3 +1575,47 @@ func TestComputeOutputsToUploadDirectories(t *testing.T) {
 		})
 	}
 }
+
+func randomBytes(n int) []byte {
+	b := make([]byte, n)
+	rand.Read(b)
+	return b
+}
+
+func BenchmarkComputeMerkleTree(b *testing.B) {
+	rand.Seed(time.Now().UnixNano())
+
+	e, cleanup := fakes.NewTestEnv(b)
+	defer cleanup()
+
+	construct(e.ExecRoot, []*inputPath{
+		{path: "a", fileContents: randomBytes(2048)},
+		{path: "b", fileContents: randomBytes(9999)},
+		{path: "c", fileContents: randomBytes(1024)},
+		{path: "d/a", fileContents: randomBytes(4444)},
+		{path: "d/b", fileContents: randomBytes(7491)},
+		{path: "d/c", emptyDir: true},
+		{path: "d/d/a", fileContents: randomBytes(5912)},
+		{path: "d/d/b", fileContents: randomBytes(9157)},
+		{path: "d/d/c", isSymlink: true, symlinkTarget: "../../b"},
+		{path: "d/d/d", fileContents: randomBytes(5381)},
+	})
+
+	inputSpec := &command.InputSpec{
+		Inputs: []string{"a", "b", "c", "d/a", "d/b", "d/c", "d/d/a", "d/d/b", "d/d/c", "d/d/d"},
+	}
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		fmc := filemetadata.NewSingleFlightCache()
+		_, _, _, err := e.Client.GrpcClient.ComputeMerkleTree(e.ExecRoot, inputSpec, fmc)
+		if err != nil {
+			b.Errorf("Failed to compute merkle tree: %v", err)
+		}
+	}
+	b.StopTimer()
+
+	for _, f := range inputSpec.Inputs {
```

Comment:
remove this and write files under b.TempDir()?
https://golang.org/pkg/testing/#B.TempDir

Reply:
Ah, `NewTestEnv` already uses a TempDir. Done.

Reply:
not yet removed?

Reply:
My bad, missed the commit when testing the benchmark before & after. Fixed.